zebra-chain/src/transaction/arbitrary.rs

```diff
@@ -177,6 +186,24 @@ impl Arbitrary for LockTime {
     type Strategy = BoxedStrategy<Self>;
 }
 
+impl Arbitrary for NetworkUpgrade {
```

Comment:
If we manually implement Arbitrary, we have to remember to update it every time we add a network upgrade.

Instead, automatically derive an Arbitrary implementation:
https://altsysrq.github.io/proptest-book/proptest-derive/getting-started.html#using-derive

In Zebra, we put `#[cfg_attr(any(test, feature = "proptest-impl"), derive(Arbitrary))]` before the type declaration. (In network_upgrade.rs.)

Reply:
I added this to the proptest docs in https://github.com/ZcashFoundation/zebra/pull/2081

Reply:
The problem here is that not al the values of the `NetworkUpgrade` are valid in a transaction. Specifically `Genesis` and `Genesis`.

Reply:
I added a comment here, will be happy to change it if there is an alternative.

https://github.com/ZcashFoundation/zebra/pull/2075/commits/a8e3578801b6ed6bdf64f23b50cf7f92b4414ab6

Reply:
Arbitrary for NetworkUpgrade needs to work anywhere we want a network upgrade - not just for v5 transactions. So it should be able to return any network upgrade value.

If you need a proptest strategy that only returns some network upgrades, then create a NetworkUpgrade method that returns a custom strategy. You could call it `branch_id_strategy()`.

I don't think there's any easy way to automatically add new network upgrades. (We could use filter on a derived Arbitrary for NetworkUpgrade, but that would reject too many values.)

Here's an example of a custom strategy:
https://github.com/ZcashFoundation/zebra/blob/75d29aca249fda0d24fcef09a99c5a858da7e0fd/zebra-chain/src/transaction/arbitrary.rs#L105